src/test/kotlin/com/personio/synthetics/e2e/EndToEndMultiStepApiTest.kt

```diff
@@ -0,0 +1,278 @@
+                        )
+
+                        assertions {
+                            statusCode(200)
+                            bodyContainsJsonPath("\$.success", "true")
+                            bodyContains("employee_id")
+                        }
+
+                        extract("PERSON_ID") {
+                            bodyJsonPath("\$.data.employee_id")
+                        }
+                    }
+                }
+            )
+
+            add(
+                apiStep("[Write API] Update Employment") {
+                    request {
+                        url("{{ BASE_URL_HRM }}/public-write-api/synthetic-monitoring/v1/persons/{{ PERSON_ID }}/employments/{{ EMPLOYMENT_ID_ANY }}")
+                        method(RequestMethod.PATCH)
+
+                        cookies("AWSELBAuthSessionCookie-0={{ ALB_COOKIE }};personio_session={{ PERSONIO_SESSION }};")
+
+                        body(
+                            """
+                            {
+                            	"position_title": "{{ EMPLOYMENT_POSITION_NEW }}"
+                            }
+                            """.trimIndent()
+                        )
+
+                        headers(
+                            mapOf(
+                                "Content-Type" to "application/json",
+                                "Accept" to "application/json",
+                                "x-csrf-token" to "{{ CSRF_TOKEN }}"
+                            )
+                        )
+
+                        assertions {
+                            statusCode(204)
+                        }
+                    }
+                }
+            )
+
+            add(
+                apiStep("[Write] Validate Person is updated") {
+                    request {
+                        url("{{ BASE_URL_HRM }}/employee-header-bff/{{ PERSON_ID }}")
+                        method(RequestMethod.GET)
+
+                        cookies("AWSELBAuthSessionCookie-0={{ ALB_COOKIE }};personio_session={{ PERSONIO_SESSION }};")
+
+                        body(
+                            """
+                            {
+                            	"position_title": "{{ EMPLOYMENT_POSITION_NEW }}"
+                            }
+                            """.trimIndent()
+                        )
+
+                        headers(
+                            mapOf(
+                                "Content-Type" to "application/json",
+                                "Accept" to "application/json",
+                                "x-csrf-token" to "{{ CSRF_TOKEN }}"
+                            )
+                        )
+
+                        assertions {
+                            bodyContainsJsonPath("\$.data.position", "{{ EMPLOYMENT_POSITION_NEW }}")
+                        }
+                    }
+                }
+            )
+        }
+    }
+
+    @Test
+    fun `create synthetic test`() {
```

Comment:
Two minor things:

- I think adding a description to this test with a brief explaining what is being tested. Even though the test code should be self explanatory, this test ties up the entire feature, so it is nice to use it as a learning resource for others.

- This test splits the logic between `@Test` and `@BeforeEach` methods, why not having everything in one `@Test` (granted, it will be a huge test, but only one).